helmchart/sdp/values.yaml

```diff
@@ -4,116 +4,178 @@
-    # If `tag` is set, it'll override the default value set in `.Chart.AppVersion`.
     tag: "latest"
 
-  # =========================== START sdp.deployment ===========================
+  ## @extra sdp.deployment Configuration related to the deployment of the SDP service.
+  ## @param sdp.deployment.annotations Annotations to be added to the deployment.
+  ## @extra sdp.deployment.podAnnotations Annotations specific to the pods, mainly for integration with monitoring tools like Prometheus.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/path The path where Prometheus metrics are exposed.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/port The port on which Prometheus metrics are exposed. This should match the SDP metrics port.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/scrape Indicates if Prometheus should scrape metrics from this pod.
+  ## @param sdp.deployment.podSecurityContext Security settings for the pods.
+  ## @param sdp.deployment.securityContext Security settings for the container within the pod.
+  ## @extra sdp.deployment.strategy Configuration related to the deployment strategy, ensuring smooth updates and minimal downtime.
+  ## @param sdp.deployment.strategy.type Specifies the deployment strategy, RollingUpdate is preferred for minimal downtime.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxUnavailable The maximum number of pods that can be unavailable during an update.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxSurge The maximum number of pods that can be created over the desired number of pods.
+  ## @param sdp.deployment.nodeSelector Node selector to determine which nodes should run the pods.
+  ## @param sdp.deployment.tolerations Tolerations to ensure pods aren't scheduled on unsuitable nodes.
+  ## @param sdp.deployment.affinity Affinity rules to determine where pods get scheduled based on node conditions.
   deployment:
     annotations:
-
     podAnnotations:
       prometheus.io/path: /metrics
       prometheus.io/port: '{{ include "sdp.metricsPort" . }}'
       prometheus.io/scrape: "true"
     podSecurityContext: { }
-    # fsGroup: 2000
     securityContext: { }
-      # capabilities:
-      #   drop:
-      #   - ALL
-      # readOnlyRootFilesystem: true
-    # runAsNonRoot: true
-    # runAsUser: 1000
     strategy:
-      # Ensure we upgrade 1 pod at a time to avoid migration races
       type: "RollingUpdate"
       rollingUpdate:
         maxUnavailable: 0
         maxSurge: 1
-
     nodeSelector: { }
-
     tolerations: [ ]
-
     affinity: { }
 
-  # =========================== START sdp.configMap ===========================
-  # ConfigMap is used to configure the SDP:
+  ## @extra sdp.configMap Configuration for the ConfigMap used by the SDP service.
+  ## @param sdp.configMap.annotations Annotations to be added to the ConfigMap.
+  ## @param sdp.configMap.data.EC256_PUBLIC_KEY [string] The EC256 public key used for authentication purposes.
```

Comment:
💡  Can we add an extra for `sdp.configMap.data` explaining that its's used to inject env vars (that are not sensitive data) for the sdp deployment?

We could also mention that the list of all configurable variables may not be 100% up to date with the application, and one can use the application's CLI's `-h` command to see the latest version of these variables that's available.

The same applies for:
- [ ] sdp.configMap.data
- [ ] anchorPlatform.configMap.data
- [ ] tss.configMap.data
- [ ] dashboard.configMap.data